### PR TITLE
Add resist & DoT-dealt proficiency activity bindings (#1338)

### LINKED ISSUE #1338: Proficiency: resist & DoT-dealt activity bindings
Part of spike **#1318** — full design in [`docs/spikes/1318-proficiency-progression-avenues.md`](../blob/main/docs/spikes/1318-proficiency-progression-avenues.md).

## What

Add the **incoming-book** and **DoT** activity bindings on top of the accrual engine:

- **Resist paths** — a path keyed to a type's *exposure* trains on `pie × (preMitigationExposure_type ÷ totalAttributes)`. Routes via the same `applies()` map on the incoming side (e.g. a fire hit trains a `Fire` resist path and an `Elemental` resist path).
- **DoT-dealt paths** — bleed/poison/burn masteries train on `pie × (dotDamageDealt_type ÷ totalAttributes)`, consuming the per-type DoT accumulators. **Type-routed, no source-skill attribution** — a bleed tick trains the Bleed path because it is Bleed-typed; this sidesteps reworking the folded per-second accumulator.

## Touches
The activity-key resolution for the incoming book, the per-type DoT-dealt aggregate (from #1320's split accumulators), and the binding/routing tests.

## Notes
- **Depends on:** the #1318 `BattleStats` additions (pre-mitigation exposure) and #1320's DoT split (typed per-second accumulators).

### PR DESCRIPTION
**Pull request by ginderjeremiah**
## Summary

Wires the two remaining damage-typed proficiency-training avenues onto the effect-based accrual engine (spike **#1318**), completing the **two-book** model alongside the offense direct-hit binding shipped in #1336:

- **Resist paths (incoming book).** A path keyed on a *resist* activity key trains on the player's **pre-mitigation exposure** to a type — `pie × clamp(exposure_type ÷ power)` — routed through the same `DamageTypes.Applies(type)` map on the incoming side, so a fire hit you take trains the **Fire-resist** path *and* the **Elemental-resist** path. Pre-mitigation, so a resist never throttles its own training signal.
- **DoT-dealt paths (offense book).** The player's typed DoT damage dealt now trains the offense Bleed/Poison/Burn (and folded Fire/Elemental/Dot) paths — **type-routed with no source-skill attribution** (a bleed tick trains Bleed because it is Bleed-typed), which sidesteps reworking the folded per-second accumulator.

Both prerequisites were satisfied before starting: #1337 (BattleStats additions) and #1325 (#1320's DoT split) are merged.

> **Rebased onto #1339** (proc/heal/reflect combat-event bindings, merged after this PR opened). Both PRs branched from #1336 and edited `BuildActivities`, the `EActivityKey` enum, and the accrual tests. The merged `BuildActivities` now folds **all three** avenue groups through one pass: offense (`TypedDamageDealt`) + resist (`TypedDamageExposure`) via `applies()`, plus the type-neutral combat events (Crit/Dodge/Heal) from #1339. #1339's `OffenseAndEventAxes` test was updated to feed offense via the converged `TypedDamageDealt`, and the duplicate `SeedPlayerAsync`/`AccrueStatsAsync` helpers were de-duplicated.

## How it addresses the issue

| Issue ask | Change |
| --- | --- |
| Resist activity bindings (incoming book) | New `EActivityKey` resist keys (`PhysicalResist`..`DotResist`) + `ActivityKeys.ForDamageKeyResist`; `BuildActivities` folds `BattleStats.TypedDamageExposure` (already tracked pre-mitigation by #1337) into the resist keys. |
| DoT-dealt aggregate (from #1320's split accumulators) | `Battler.ApplyDamageOverTime` gains a post-mitigation per-type damage-dealt recorder; `BattleContext` wires the enemy's DoT into `TypedDamageDealt`. |
| Activity-key resolution + binding/routing tests | One shared fold routes both books; resist/offense keys are disjoint. Unit + integration coverage added. |

### Notable design decision

#1337 pre-tracked two books — `TypedDamageDealt` (output) and `TypedDamageExposure` (incoming) — for the accrual to consume. #1336 wired offense via a `SkillStats` re-derivation (equivalent for direct hits, since a hit's type is always its skill's `DamageType`) and left `TypedDamageDealt` **unread**. This PR **converges offense onto `TypedDamageDealt`** — now the single typed-dealt book carrying direct hits *and* DoT — and adds the resist binding consuming `TypedDamageExposure`. That removes the redundant per-skill re-derivation and the dead field rather than introducing a second parallel DoT aggregate. Offense XP for direct hits is unchanged (the one exception is a pathological absorption edge case — enemy resist >100% with two same-type skills — where net-negative same-type damage now nets in rather than being dropped per-skill, which is arguably more correct). Flagging it explicitly since it touches just-merged #1336 code.

The DoT-dealt recorder, like the exposure recorder, is a backend-only side channel that never touches the health math — **no new battle-parity surface** (accrual is off the deterministic tick loop; `BattleStats` is backend-only).

## Test plan

- `dotnet build` — clean; `dotnet format --verify-no-changes` — clean.
- Backend tests (post-rebase): **Core 1013**, **Application 640**, **Api 646** (codegen-drift incl. regenerated `enums.ts`), **Infrastructure 60** — all green.
- Frontend: `npm run lint` (eslint + svelte-check) clean; proficiency/progression unit tests (162) green.
- New/updated coverage:
  - `ActivityKeysTests` — each damage key → its offense + resist key; offense/resist disjoint.
  - `BattleDamageOverTimeTests` — DoT-dealt recorder reports post-mitigation per type; exposure vs. dealt split by side; DoT-on-enemy now type-routed into `TypedDamageDealt`.
  - `ProficiencyRewardServiceTests` — resist routing (Fire exposure → Fire/Elemental-resist, not Fire offense), two-book parallel training, DoT-dealt routing (Bleed → Bleed + Dot, not Fire); offense integration tests updated to the converged `TypedDamageDealt` source.

## Docs

Updated `game-design.md` (routing now spans both books; offense + resistance + the type-neutral crit/dodge/heal events wired, only Retribution/reflect remaining) and `backend.md` (the accrual consumes `TypedDamageDealt`/`TypedDamageExposure`, folded to offense/resist keys).

## Follow-ups (unchanged scope)

- Retribution (reflected damage) remains the only unwired avenue, pending the mitigation rework **#1330**. Crit/dodge/heal landed in **#1339** (now merged in).
- Friendly activity-key labels and admin authoring remain **#1341** (the client still renders raw enum names via the existing `TODO(#1341)`). Resist paths don't show the loadout-derived "training" pulse, which is keyed off offense skills — a UI legibility concern, not accrual.

Closes #1338

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: Game.Core/Battle/Battler.cs [hunk 1]
```diff
@@ -173,7 +173,18 @@ public double TakeDamage(double dealt, EDamageType damageType, double blockReduc
...
+        /// invoked with each DoT type and the tick damage <em>after</em> this battler's resistance, the same
+        /// value the tick subtracts from health. Supplied when this battler is the victim of the player's DoT
+        /// (the enemy), so the player's DoT damage dealt is typed for the offense binding consistently with a
+        /// direct hit's post-mitigation actual damage; <c>null</c> leaves the loop unchanged. Like
+        /// <paramref name="recordExposure"/> it is a backend-only side channel that adds no parity surface.
+        /// </param>
+        public double ApplyDamageOverTime(
+            int ms,
+            Action<EDamageType, double>? recordExposure = null,
+            Action<EDamageType, double>? recordDamageDealt = null)
```
Comment (ginderjeremiah) [inline Game.Core/Battle/Battler.cs:187]: Nit (footgun): two trailing optional params of the same type (`Action<EDamageType, double>?`) make it easy for a future caller to land an exposure recorder in the dealt slot, or vice-versa, with no compiler complaint. The current call sites are safe — the player-exposure call passes positionally and the dealt call uses a named arg (`recordDamageDealt:`) — so this is fine as-is. Worth keeping the named-arg convention for any new caller; no change required.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: Game.Application/Services/ProficiencyRewardService.cs [hunk 2]
```diff
@@ -159,42 +162,25 @@ private bool AllPrerequisitesMaxed(int gatedId, PlayerProgress progress)
...
             var activityByKey = new Dictionary<EActivityKey, double>();
-            foreach (var (type, damage) in damageByType)
-            {
-                foreach (var key in DamageTypes.Applies(type))
-                {
-                    var activityKey = ActivityKeys.ForDamageKey(key);
-                    activityByKey[activityKey] = activityByKey.GetValueOrDefault(activityKey) + damage;
-                }
-            }
+            FoldIntoActivityKeys(activityByKey, stats.TypedDamageDealt, ActivityKeys.ForDamageKey);
```
Comment (ginderjeremiah) [inline Game.Application/Services/ProficiencyRewardService.cs:187]: Nit (description accuracy, not the code): the PR says offense XP for direct hits is "unchanged" by the convergence. That's exact in the common case — the same `actualDamage` feeds both `SkillStats.TotalDamage` and `TypedDamageDealt[Skill.DamageType]` (`BattleSkill.Update`), so per-type sums match. The one divergence is the absorption edge case: the old per-skill loop dropped a whole skill whose net total was `<= 0`, whereas folding `TypedDamageDealt` directly nets a same-type negative against same-type positives before training. It requires an enemy resist `> 100%` plus two same-type skills, so it's pathological, and the new behavior (train on net damage actually dealt) is arguably the more correct one. Flagging only so the "unchanged" claim is precise — no change needed.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by ginderjeremiah — COMMENTED** [review]
## Review: Resist & DoT-dealt proficiency activity bindings (#1338)

**Verdict:** Approve _(submitted as a comment — I'm the PR author, so GitHub won't take a gating event from me)_ — completes the two-book model cleanly, the offense convergence onto `TypedDamageDealt` is provably equivalent for direct hits, and coverage is thorough. Only two non-blocking nits.

### Summary of what I checked

- **Offense convergence is sound.** Replacing the per-skill re-derivation with a direct fold of `TypedDamageDealt` is equivalent for direct hits: in `BattleSkill.Update` the same `actualDamage` returned by `DamageTarget` is booked into both `SkillStats[id].TotalDamage` and `TypedDamageDealt[Skill.DamageType]`, and the old code grouped `SkillStats.TotalDamage` by that very `DamageType`, so per-type sums match. This also removes the now-dead re-derivation rather than leaving it alongside a second DoT aggregate — good cleanup per CLAUDE.md. `ISkills` remains justified (skill-grant path), so no orphaned dependency.
- **Non-positive activity is handled.** Folding can produce a zero/negative key total, but the downstream `xpGain <= 0` skip in `AccrueAndApply` banks nothing for it, so no information-free rows or negative training.
- **Two books stay disjoint.** `FoldIntoActivityKeys` accumulates both into one map; offense vs. resist keys are guaranteed non-overlapping (`ActivityKeysTests.OffenseAndResistKeys_AreDisjoint`), so the shared accumulator can't cross-train axes.
- **No new battle-parity surface.** The DoT-dealt recorder computes `tickDamage` once for both the health subtraction and the record (no drift), and is a backend-only side channel. The frontend simulation tracks no `TypedDamage*` at all (accrual is server-only, off the deterministic tick loop), so there's correctly nothing to mirror.
- **Enum ordinals preserved.** Resist keys appended at 14–23, leaving the persisted `Path.ActivityKey` ordinals untouched; `enums.ts` regenerated to match.
- **Book split is clean.** `BattleContext.ResolveDamageOverTime` records the enemy's DoT as the player's typed damage-dealt and the player's incoming DoT as exposure, with no leakage — verified by `ResolveDamageOverTime_RecordsPlayerDotExposure_AndEnemyDotAsDamageDealt`.

### Nits (non-blocking, see inline)

- `ApplyDamageOverTime` now has two trailing optional params of the same delegate type — a mild footgun mitigated by the named-arg convention at the dealt call site.
- The "offense XP unchanged" claim is exact except in a pathological absorption edge case (enemy resist > 100% + two same-type skills), where net-negative same-type damage now nets in rather than being dropped per-skill — arguably more correct. Flagged only for precision.

### Test coverage

Strong and aligned with the project's rules:
- `ActivityKeysTests` (new) — each damage key → its offense + resist key, and offense/resist disjointness.
- `BattleDamageOverTimeTests` — post-mitigation DoT-dealt recorder, active-types-only, and the per-side exposure/dealt split.
- `ProficiencyRewardServiceTests` — resist routing (exposure → resist keys, not offense), two-book parallel training, and DoT-dealt type-routing with no skill attribution; offense tests updated to the converged source.
- No frontend mirror required (backend-only, off the parity surface).

### What's good

Genuinely tidy increment: one shared fold serves both books, the convergence removes redundancy instead of adding a parallel aggregate, and the design decision touching just-merged #1336 is called out explicitly. Docs (`game-design.md`, `backend.md`) and codegen are kept in sync.

### Note

Not merging despite the approve: CI reports no checks (status `pending`, zero check runs) so green can't be confirmed, and there are open nits. Follow-ups (#1339 combat-event bindings, #1341 friendly labels) are already tracked, so nothing new to file.

---
_Generated by [Claude Code](https://claude.ai/code)_
